lib/android/app/src/main/java/com/wix/reactnativenotifications/core/notification/PushNotification.java

```diff
@@ -59,7 +59,9 @@ protected PushNotification(Context context, Bundle bundle, AppLifecycleFacade ap
 
     @Override
     public void onReceived() throws InvalidNotificationException {
-        postNotification(null);
+        if (!mAppLifecycleFacade.isAppVisible()) {
+            postNotification(null);
+        }
```

Comment:
We need to notify the JS side for both background and foreground mode. Is there a reason for this change I'm not aware of?

```suggestion
        postNotification(null);
```

Reply:
from what I remember `postNotification` does not notify the JS side, but rather create and display a notification in the notifications drawer of the phone. This [issue](https://github.com/wix/react-native-notifications/issues/80) describes the problem for why I changed this. In short, we already get notified of the incoming notification on the JS side (see that on the next line `notifyReceivedToJS()` is called outside of the `if (!mAppLifecycleFacade.isAppVisible())`).

Another thing may be that this (to display an empty notification in the drawer) is intended behaviour when the notification was received while the app was opened. Maybe we should make this configurable, I think that are cases when you want to process the notification directly on the JS side without displaying anything and then choosing whether to show it (possibly using a localNotification).

And another problem (for me at least) is that `postNotification(null)` displays an empty notification. It would have been fine if the notification had the title/body/icon received from firebase, but it is rather a white, empty notification, which to me sounds like a bug.

I will remove it if you still think this is not necessary.

Reply:
@alexbuicescu @yogevbd  Can we have this optional? It's very critical for me to get notifications in a foreground mode.